akka-actor/src/main/scala/akka/pattern/CircuitBreaker.scala

```diff
@@ -176,12 +252,28 @@ class CircuitBreaker(
    * @return [[java.util.concurrent.CompletionStage]] containing the call result or a
    *   `scala.concurrent.TimeoutException` if the call timed out
    */
-  def callWithCircuitBreakerCS[T](body: Callable[CompletionStage[T]]): CompletionStage[T] =
+  def callWithCircuitBreakerCS[T](
+    body: Callable[CompletionStage[T]]): CompletionStage[T] =
     FutureConverters.toJava[T](callWithCircuitBreaker(new Callable[Future[T]] {
       override def call(): Future[T] = FutureConverters.toScala(body.call())
     }))
 
   /**
+   * Java API (8) for [[#withCircuitBreaker]]
+   *
+   * @param body Call needing protected
+   * @param defineFailureFn function that define what should be consider failure and thus increase failure count
+   * @return [[java.util.concurrent.CompletionStage]] containing the call result or a
+   *   `scala.concurrent.TimeoutException` if the call timed out
+   */
+  def callWithCircuitBreakerCS[T](
+    body:            Callable[CompletionStage[T]],
+    defineFailureFn: Try[T] ⇒ Boolean): CompletionStage[T] =
```

Comment:
Java API should use `java.util.Function`, and `Try` is rather alien in Java API. Not sure what we shall use?